android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -830,9 +840,16 @@ private MediaSource buildMediaSource(Uri uri, String overrideExtension, DrmSessi
                 );
                 break;
             case CONTENT_TYPE_OTHER:
-                mediaSourceFactory = new ProgressiveMediaSource.Factory(
```

Comment:
I don't understand why you have to change the 'other' case, and not the hls/dash one ? Is it supported only on this 'other' case ?

Reply:
tbh I don't know much about android native; just adding a simple cache DataSource Factory gives an hls malformed url error, progressiveMediaSource works fine.